### DISCUSSION TITLE
机器人回复qq消息就会发生线程阻塞

### DISCUSSION THREAD

**Discussion by Baishou-Kl** [simple-robot/Discussions#38] [Q&A]
服务器信息: 共享ip win10系统  ,未安装任何jre或jdk 
exe4j打包的exe程序程序打包时导入的jre运行环境
程序启动起来后 能够监听到群聊消息,但机器人无法发送消息 且还会阻塞当前线程,打包好的程序在个人电脑上又能够正常运行,无任何异常
目前并不能找到问题发生原因
在前一段时间也有过同样的情况 能够监听到qq消息 但机器人回复消息时就无法正常发送消息且阻塞当前线程,当时也是服务器上运行,使用的是jar包运行,win 2012 server 系统

**Comment by ForteScarlet** [comment]
参考 [这个问题](https://github.com/simple-robot/simbot-component-kook/issues/62) ，尝试更新版本

**Reply by Baishou-Kl** [reply]
了解,再次麻烦大佬了

**Comment by Baishou-Kl** [comment]
使用新版本M5之后上述问题得到解决,但貌似 bot-configuration-resources 这个扫描配置失效了
![image](https://user-images.githubusercontent.com/89904695/206912409-99334324-cf91-4f52-a395-398942ae5646.png)

**Reply by ForteScarlet** [reply]
在 Spring Boot Starter 中，这个配置项基于Spring的 `PathMatchingResourcePatternResolver`，尝试使用 `classpath:simbot-bots/*.bot*` 或 `file:simbot-bots/*.bot*`

或者观察启动日志，查看是否有失败、忽略等原因的相关日志

**Reply by Baishou-Kl** [reply]
了解大佬 早些许休息吧
